### PR TITLE
Add environment table component

### PR DESCRIPTION
**Pull request by Flanker32**
- Add environment table component
- Align Spring deployment configuration component

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/springcloud/ui/EnvironmentVariablesTextFieldWithBrowseButton.java [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+
+import javax.swing.*;
+import javax.swing.event.ChangeEvent;
+import javax.swing.event.ChangeListener;
+import javax.swing.event.DocumentEvent;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+public class EnvironmentVariablesTextFieldWithBrowseButton extends TextFieldWithBrowseButton
```
Comment (andxu) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/springcloud/ui/EnvironmentVariablesTextFieldWithBrowseButton.java:44]: We can use EnvironmentVariableTable only without text field in setting edtor.

Comment (Flanker32) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/springcloud/ui/EnvironmentVariablesTextFieldWithBrowseButton.java]: Removed this component and use `EnvironmentVariableTable` directly

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/springcloud/ui/EnvironmentVariableTable.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+                environmentVariables.keySet().stream()
+                                    .map(key -> new EnvironmentVariable(key, environmentVariables.get(key), true))
+                                    .collect(Collectors.toList());
+        setValues(environmentVariableList);
+    }
+
+    public Map<String, String> getEnv() {
+        Map<String, String> result = new LinkedHashMap<>();
+        for (EnvironmentVariable variable : this.getEnvironmentVariables()) {
+            if (StringUtil.isEmpty(variable.getName()) && StringUtil.isEmpty(variable.getValue())) {
```
Comment (andxu) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/springcloud/ui/EnvironmentVariableTable.java:57]: if (StringUtil.isEmpty(variable.getName()) continue;

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/springcloud/ui/SpringCloudAppSettingPanel.java [hunk 3]
```diff
@@ -480,4 +486,14 @@ protected JLabel getLblArtifact() {
     protected JLabel getLblMavenProject() {
         return lblMavenProject;
     }
+
+    private void createUIComponents() {
+        // TODO: place custom component creation code here
```
Comment (andxu) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/springcloud/ui/SpringCloudAppSettingPanel.java:491]: remove this comment

**Pull request merged**